src/components/news-list/news-list.js

```diff
@@ -1,51 +1,79 @@
-import React, { Component } from "react";
+import React, { useEffect, useState } from "react";
 
 import NewsService from "../../services/news-service";
 import NewsListItem from "../news-list-item";
 import "./news-list.css";
 
-// import Spinner from "../spinner/spinner";
+import Spinner from "../spinner/spinner";
+import Pagination from "../pagination";
+const newsService = new NewsService();
 
-export default class NewsList extends Component {
-  newsService = new NewsService();
+const NewsList = () => {
+  const [newsIDBestNews, setnewsIDBestNews] = useState([]);
+  const [loading, setloading] = useState(false);
+  const [firstItemOnPage, setfirstItemOnPage] = useState(0);
```

Comment:
It makes sense to simplify pagination logic. For instance, separate items indexes to show and the current page state.
To replace `firstItemOnPage` with `currentPage` state.
For example, some parts of the code.
```ts
const [currentPage, setCurrentPage] = useState(0);
const onPreviousPage = () => setCurrentPage(page => page - 1);
const onNextPage = () => setCurrentPage(page => page + 1);
const currentItemIds = useState([]);
```

And the logic for items to show:

```js

const [itemsToShow, setItemsToShow] = useState([]);
// Update items to show if page changed or newsIDBestNews updated
useEffect(() => {
    const itemsPerPage = 20;
    const startIndex = currentPage * itemsPerPage;
    const endIndex = startIndex + itemsPerPage;
    setItemsToShow(newsIDBestNews.slice(startIndex, endIndex));
}, [newsIDBestNews, currentPage, setItemsToShow]);
```

Reply:
const currentItemIds = useState([]);
What is it?